### PR TITLE
Update minimum compiler version

### PR DESCRIPTION
**Pull request by erick-xanadu**
**Description of the Change:** The minimum compiler version needs to be updated due to device API update. This is the last PR to be merged to the release candidate as it depends on Catalyst being updated.

### CODE REVIEW

**Review by trbromley — APPROVED** [review]
Tests will still pass even though `0.6.0` isn't yet available?

**Review by AmintorDusko — APPROVED** [review]

**Comment by erick-xanadu** [comment]
@trbromley mmm.... probably not. This could also be merged until after we release 0.6.0?

**Comment by erick-xanadu** [comment]
This is not required! Sorry for the confusion.

**Comment by erick-xanadu** [comment]
Reopening. This is needed because of lightning.

:heavy_check_mark:  pennylane==0.36.0, catalyst==0.5.0, lightning==0.35.1
:x:  pennylane==0.36.0, catalyst==0.5.0, lightning>0.35.1
:heavy_check_mark:  pennylane==0.36.0, catalyst==0.6.0, lightning==0.36.0

File: tests/test_compiler.py [hunk 1]
```diff
@@ -43,7 +43,7 @@ def catalyst_incompatible_version():
 
 @pytest.mark.usefixtures("catalyst_incompatible_version")
 def test_catalyst_incompatible():
-    """Test qjit with an incompatible Catalyst version < 0.4.0"""
+    """Test qjit with an incompatible Catalyst version < 0.6.0"""
```
Comment (dime10) [inline tests/test_compiler.py:46]: We should generalize this so it doesn't need updating every time
```suggestion
    """Test qjit with an incompatible Catalyst version that's lower than required."""
```

File: tests/test_compiler.py [hunk 2]
```diff
@@ -53,7 +53,7 @@ def circuit():
         return qml.state()
 
     with pytest.raises(
-        CompileError, match="PennyLane-Catalyst 0.5.0 or greater is required, but installed 0.0.1"
+        CompileError, match="PennyLane-Catalyst 0.6.0 or greater is required, but installed 0.0.1"
```
Comment (dime10) [inline tests/test_compiler.py:56]: Not sure if regex or something is allowed here? Alternatively I usually just truncate the matched error message, but here the version is in the middle 🤔

**Review by dime10 — COMMENTED** [review]
👍

**Review by jay-selby — APPROVED** [review]

**Pull request merged**